aggregator/src/aggregator/collection_job_driver.rs

```diff
@@ -217,16 +247,20 @@ impl CollectionJobDriver {
 
                     match maybe_updated_collection_job.state() {
                         CollectionJobState::Start => {
-                            tx.update_collection_job::<SEED_SIZE, Q, A>(&collection_job).await?;
-                            tx.release_collection_job(&lease).await?;
+                            try_join!(
+                                tx.update_collection_job::<SEED_SIZE, Q, A>(&collection_job),
+                                try_join_all(batch_aggregations.iter().map(|ba| tx.update_batch_aggregation(ba))),
```

Comment:
I'm concerned that this update could stomp on concurrent changes from a concurrent aggregation flow, running between the `step_collection_job_1` and `step_collection_job_2` transactions. The intent of this update is to flip the batch aggregation's state enum, but if the aggregate share/report count/checksum/timestamp interval have been updated in the database, we'd lose those changes, when we should instead retry in hopes of getting a consistent view of the batch. Our leasing system only protects changes to the collection job, not batch aggregations. (the below `put_batch_aggregation()` call should be okay, because any concurrently created batch aggregation rows would result in an error, and an eventual retry at the job driver level)

Reply:
I moved the update of the batch aggregation state to the first transaction to avoid this possibility.